packages/suite/src/actions/suite/messageSystemActions.ts

```diff
@@ -0,0 +1,129 @@
+          payload: MessageSystem;
+          isRemote: boolean;
+      }
+    | { type: typeof MESSAGE_SYSTEM.FETCH_CONFIG_ERROR }
+    | {
+          type: typeof MESSAGE_SYSTEM.SAVE_VALID_MESSAGES;
+          category: Category;
+          payload: string[];
+      }
+    | {
+          type: typeof MESSAGE_SYSTEM.DISMISS_MESSAGE;
+          category: Category;
+          id: string;
+      };
+
+const fetchSuccess = (): MessageSystemAction => ({
+    type: MESSAGE_SYSTEM.FETCH_CONFIG_SUCCESS,
+});
+
+const fetchSuccessUpdate = (payload: MessageSystem, isRemote: boolean): MessageSystemAction => ({
+    type: MESSAGE_SYSTEM.FETCH_CONFIG_SUCCESS_UPDATE,
+    payload,
+    isRemote,
+});
+
+const fetchError = (): MessageSystemAction => ({
+    type: MESSAGE_SYSTEM.FETCH_CONFIG_ERROR,
+});
+
+const fetchConfig = () => async (dispatch: Dispatch, getState: GetState) => {
+    const { timestamp, currentSequence } = getState().messageSystem;
+
+    if (Date.now() >= timestamp + FETCH_INTERVAL) {
+        try {
+            let jwsResponse;
+            let isRemote = true;
+
+            try {
+                const response = await fetch(CONFIG_URL_REMOTE);
+
+                if (!response.ok) throw Error(response.statusText);
+
+                jwsResponse = await response.text();
+            } catch (error) {
+                console.error(`Fetching of remote JWS config failed: ${error}`);
+
+                const response = await fetch(CONFIG_URL_LOCAL);
+
+                if (!response.ok) {
+                    throw Error(`Fetching of local JWS config failed: ${response.statusText}`);
+                }
+
+                jwsResponse = await response.text();
+                isRemote = false;
+            }
+
+            const decodedJws = jws.decode(jwsResponse);
+
+            if (!decodedJws) {
+                throw Error('Decoding of config failed');
+            }
+
+            const { alg } = decodedJws?.header;
+            // Disable eslint rule as object destruction does not work here
+            // eslint-disable-next-line prefer-destructuring
+            const PUBLIC_KEY = process.env.PUBLIC_KEY;
+
+            const isAuthenticityValid = jws.verify(jwsResponse, alg, PUBLIC_KEY!);
+
+            if (!isAuthenticityValid) {
+                throw Error('Config authenticity is invalid');
+            }
+
+            const config: MessageSystem = JSON.parse(decodedJws.payload);
+
+            if (VERSION !== config.version) {
+                throw Error('Config version is not supported');
+            }
+
+            if (currentSequence < config.sequence) {
+                await dispatch(fetchSuccessUpdate(config, isRemote));
+            } else if (currentSequence === config.sequence) {
+                await dispatch(fetchSuccess());
+            } else {
+                throw Error('Sequence of config is older than the current one');
+            }
+        } catch (error) {
+            console.error(error);
```

Comment:
console.error will be logged in sentry. I'm just afraid nobody will notices anyway, but still could be useful to determine how many users are affected 😁

Reply:
> console.error will be logged in sentry.

Perfect! Good to know.

> I'm just afraid nobody will notices anyway,

Yeah. Reporting to sentry is one thing. Getting these errors in front of the right eyes is another story. .D Where do sentry errors currenrly end up? At GoOut we had a dedicated slack channels where sentry (and other runtime) errors were posted by a bot. Shall I create an issue for such an integration?

I'll resolve this thread as the original concren is addressed.

Reply:
Reports are only in sentry right now. Some kind of bot is interesting idea, but since we have literally thousands of reports so it could get ugly 😁 Maybe it could report only after certain amount of people experienced the issue or something like that, that would be really cool!

Imo first we need address these reports. So time ago I created github issues from few most common sentry reports (https://github.com/trezor/trezor-suite/issues?q=is%3Aopen+is%3Aissue+label%3ASentry). Another problem is that too many people are using old version  of the app (because of non functioning auto updater) and it is kinda hard to organise all these issues (I blame sentry UI, but it could be that we are just doing something wrong). Anyway It will take quite a lot of time and effort to clean up sentry to it can really start useful and sadly priorities are elsewhere a the moment 😞

Reply:
Aaaah, so we need to get out of the dirty waters first to actually see through to valuable information.

> Maybe it could report only after certain amount of people experienced the issue or something like that, that would be really cool!

At first sight probably yes. At second I'd call that a band aid at best. We're not a service distributed across datacenters all over the globe to ignore erros not reaching an occurence threshold. (OFC occurence should still be used for triage.) I think that we need to aim for 0 errors reported. The closer we get to that the more valuable the reported errors will be. That goal is impossibl of course. But that doesn't stop us from aiming for it. ,)

> Imo first we need address these reports. So time ago I created github issues from few most common sentry reports

This is IMO a perfect start. 👍 I think it wouldn't be a bad idea to include these issues in the Acceptence criteria of Suite official start.

Btw: how do I access the Sentry reports?